compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -406,15 +412,45 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
 
         val altsByTypeTest = groupByWithoutHashCode(alts) {
           case ExportedSymbol(alt) =>
-            // get parameter type while resolving repeated params
+            // See the comment in genPrimitiveJSArgs for a rationale about this
+
             val tpe = enteringPhase(currentRun.uncurryPhase) {
-              val ps = alt.paramss.flatten
-              if (ps.size <= paramIndex || isRepeated(ps(paramIndex))) {
-                assert(isRepeated(ps.last))
-                repeatedToSingle(ps.last.tpe)
+
+              lazy val paramsTypesUncurry = alt.paramss.flatten.map(_.tpe)
+              lazy val isRepeatedUncurry = alt.paramss.flatten.map(isRepeated)
+
+              def paramTypesPosterasure = enteringPhase(currentRun.posterasurePhase) {
+                alt.paramss.flatten.apply(paramIndex).tpe
```

Comment:
That one exists but makes the compiler crash. I looks like those parameters are not the ones in `enteringPhase(currentRun.uncurryPhase)`.